lib/resty/mlcache.lua

```diff
@@ -170,25 +172,48 @@ function _M.new(shm, opts)
 
         self.ipc:subscribe(self.ipc_invalidation_channel, function(key)
             self.lru:delete(key)
+
+            for i = 1, #self.lrus do
+                self.lrus[i]:delete(key)
+            end
         end)
     end
 
     return setmetatable(self, mt)
 end
 
 
-local function set_lru(self, key, value, ttl)
+function _M:add_lru(name, lru)
+    if type(name) ~= "string" or #name == 0 then
+        return error("name must be a non-empty string")
+    end
+
+    if type(lru) ~= "table" then
+        return error("lru appears not to be a lua-resty-lrucache instance")
+    end
+
+    if self.lrus[name] then
+        return error("an lru named '" .. name .. "' has already been added")
+    end
+
+    self.lrus[name] = lru
+
+    insert(self.lrus, lru)
```

Comment:
style: insert() here, but elsewhere #self.lrus. why not `self.lrus[#self.lrus] = lru` here?

related: do we care enough about micro-optimizations to store an lrus table index as part of self, rather than than recomputing `#lrus`